src/web/extractors.rs

```diff
@@ -46,18 +42,49 @@ pub struct UidParam {
     uid: u64,
 }
 
-#[derive(Debug, Deserialize, Serialize)]
+#[derive(Debug, Deserialize, Serialize, Validate)]
 pub struct BatchBsoBody {
+    #[validate(custom = "validate_body_bso_id")]
     pub id: String,
+    #[validate(custom = "validate_body_bso_sortindex")]
     pub sortindex: Option<i32>,
     pub payload: Option<String>,
+    #[validate(custom = "validate_body_bso_ttl")]
     pub ttl: Option<u32>,
 }
 
+impl BatchBsoBody {
+    /// Function to convert valid raw JSON BSO body to a BatchBsoBody
+    fn from_raw_bso(val: &Value) -> Option<BatchBsoBody> {
+        let map = match val.as_object() {
+            Some(m) => m,
+            None => return None,
+        };
+        // Verify all the keys are valid
+        let valid_keys = ["id", "sortindex", "payload", "ttl"];
+        for key_name in map.keys() {
```

Comment:
nit this could be a one liner any()

Reply:
I guess I'm feeling dense as I'm not seeing what would be needed to make that happen.